inc/updater/namespace.php

```diff
@@ -35,32 +35,92 @@ function get_packages() {
 	$plugin_path = trailingslashit( WP_PLUGIN_DIR );
 	$plugins     = get_plugins();
 	foreach ( $plugins as $file => $plugin ) {
-		if ( empty( $plugin['UpdateURI'] ) ) {
-			continue;
-		}
 		$plugin_id = get_file_data( $plugin_path . $file, [ 'PluginID' => 'Plugin ID' ] )['PluginID'];
-
 		if ( ! empty( $plugin_id ) ) {
-			$packages['plugins'][] = $plugin_path . $file;
+			$packages['plugins'][ $plugin_id ] = $plugin_path . $file;
 		}
 	}
 
 	$theme_path = WP_CONTENT_DIR . '/themes/';
 	$themes     = wp_get_themes();
 	foreach ( $themes as $file => $theme ) {
-		if ( empty( $theme->get( 'UpdateURI' ) ) ) {
-			continue;
-		}
 		$theme_id = get_file_data( $theme_path . $file . '/style.css', [ 'ThemeID' => 'Theme ID' ] )['ThemeID'];
-
 		if ( ! empty( $theme_id ) ) {
-			$packages['themes'][] = $theme_path . $file . '/style.css';
+			$packages['themes'][ $theme_id ] = $theme_path . $file . '/style.css';
 		}
 	}
 
 	return $packages;
 }
 
+/**
+ * Get icons.
+ *
+ * @param  array $icon Array of icon data.
+ *
+ * @return array
+ */
+function get_icons( $icon ) {
+	if ( empty( $icon ) ) {
+		return;
+	}
+
+	$icons_arr = [];
+	$icons = $icon;
+
+	$regular = array_find( $icons, fn ( $icon ) => $icon->width === 772 && $icon->height === 250 );
+	$high_res = array_find( $icons, fn ( $icon ) => $icon->width === 1544 && $icon->height === 500 );
+
+	foreach ( $icons as $icon ) {
+		foreach ( $icon as $mime => $type ) {
+			if ( $mime === 'content-type' ) {
+				if ( str_contains( $type, 'svg+xml' ) ) {
+					$svg = $icon;
+					break;
+				}
+			}
+		}
+	}
+
+	if ( empty( $regular ) && empty( $high_res ) && empty( $svg ) ) {
+		return;
+	}
+
+	$icons_arr['1x'] = $regular->url ?? '';
+	$icons_arr['2x'] = $high_res->url ?? '';
+	$icons_arr['svg'] = $svg->url ?? '';
+
+	return $icons_arr;
+}
+
+/**
+ * Get banners.
+ *
+ * @param  array $banner Banner data.
+ *
+ * @return array
+ */
+function get_banners( $banner ) {
+	if ( empty( $banner ) ) {
+		return;
```

Comment:
```suggestion
		return [];
```
Docblock says it should return an array.